private_sharing/forms.py

```diff
@@ -313,6 +315,35 @@ class UploadDataFileBaseForm(forms.Form):
 
     metadata = forms.CharField(label="Metadata", required=True)
 
+    # Note that this will need to be set to required=True to enable future enforcement
+    datatypes = forms.CharField(label="Data type", required=False)
+
+    def clean_datatypes(self):
+        """
+        Takes incoming string, converts to a sorted set of ints and does some error checking.
+        """
+        datatypes = self.cleaned_data["datatypes"]
+        # Note:  To be removed when strict datatypes are required
+        if not datatypes:
+            return set()
+        if not (datatypes.startswith("[") and datatypes.endswith("]")):
+            raise forms.ValidationError(
+                "A list of datatypes is required to describe this file"
+            )
+        # Sort the requested IDs and turn them into ints so that we match what will
+        # come out of the database
+        datatypes_list = ast.literal_eval(datatypes)
```

Comment:
It seems like it would be safer to use a data serialization format like JSON? (I didn't realize this was even possible - so that's interesting to learn, but...)